### PR TITLE
CNFT2-624 Save Page As Draft

### PR DESCRIPTION
**Pull request by Emelias-Enquizit**

### CODE REVIEW

File: apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/page/PageStateChanger.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+				response.setStatus(HttpStatus.NOT_FOUND);
+				response.setTemplateId(id);
+			}
+		} catch (Exception e) {
+			response.setMessage(e.getMessage());
+			response.setStatus(HttpStatus.INTERNAL_SERVER_ERROR);
+			response.setTemplateId(id);
+
+		}
+		return response;
```
Comment (mpeels) [inline apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/page/PageStateChanger.java:47]: Legacy does not allow converting an existing "published" page back into a draft. It creates a new page from the existing one without modifying the published page. (Plz double check and validate this though).

Workflow that I have seen:
1. Create a new page (`template_type` is set to "Draft", `publish_version_nbr` of NULL)
2. Publish that page (`template_type` is set to "Published")
3. Create a draft from a published page - creates a new entry in the `wa_template` table with status of "Draft" and a `publish_version_nbr` of 0, the published page has its `template_type` set to `Published With Draft`.

The query that fetches the pages to display their summaries in table format only selects `template_type` of "Draft" or "Published", this prevents the duplicates from showing up in the summary table.

**Review by mpeels — CHANGES_REQUESTED** [review]

File: apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/page/PageStateChanger.java (hunk 1 above)
Comment (Emelias-Enquizit) [reply apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/page/PageStateChanger.java:47]: @mpeels  sure I will confirm

Comment (Emelias-Enquizit) [reply apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/page/PageStateChanger.java]: Hi @mpeels  updated please take a look:

Screenshot 2023-08-02 at 2 29 18 PM

Screenshot 2023-08-02 at 2 47 58 PM

File: apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/page/PageStateChanger.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	}
+	
+	public WaTemplate createDraftCopy(WaTemplate oldPage) {
+		if (oldPage.getTemplateType().equals("Draft")) {
+			return oldPage;
+		}
+		Long id = templateRepository.getMaxTemplateID();
```
Comment (mpeels) [inline apps/question-bank/src/main/java/gov/cdc/nbs/questionbank/page/PageStateChanger.java:54]: The id column for `Wa_template` will auto generate an Id.
```java
@Id
@GeneratedValue(strategy = GenerationType.IDENTITY)
@Column(name = "wa_template_uid", nullable = false)
private Long id;
```

**Review by mpeels — APPROVED** [review]

**Pull request merged**